gradio/blocks.py

```diff
@@ -339,6 +353,8 @@ async def process_api(
         block_fn = self.fns[fn_index]
         dependency = self.dependencies[fn_index]
 
+        await self.set_thread_limit()
```

Comment:
Why do we `set_thread_limit` in `process_api`, let's do it within launch?
If the running async function in a sync function is a problem we can handle it with anyio or asyncio, let me know if this is the case.

Reply:
I agree, it's better in `launch`, I'll take a look. Also, we may have to store the original `run_sync` if we need to support multiple `launch` invocations instead of nesting a lot of partials. This patching approach does not feel right, do you have any other suggestions?

Reply:
This [could](https://github.com/gradio-app/gradio/pull/1443#discussion_r887802872) work!

Reply:
I just committed dfa1c56d8bbf33636cc5c878fd4bc774d2a69859.

I'll try to setup the test environment to see why circleci is failing.